CougSat1-PMIC/src/events/EventPeriodic.cpp

```diff
@@ -17,9 +17,629 @@
 #include "tools/CISConsole.h"
 #include "tools/CISError.h"
 
+
 /**
  * @brief Executes the periodic event
  *
  * @return uint8_t error code
  */
-uint8_t eventPeriodic() { return ERROR_NOT_SUPPORTED; }
\ No newline at end of file
+uint8_t eventPeriodic()
+{
+
+	// From powerrail psuedocode, implement control algorithm
+	// Control input switching to power
+
+	// Monitor battery
+	long BattVoltA = 0; // voltage level of battery A
+	long BattVoltB = 0; // Voltage level of battery B
+	long BattCurrA = 0; // Current leaving Battery A
+	long BattCurrB = 0; // Current Leaving Battery B
+	long PLCurr = 0; // Temp variable that will hold a current level value
+	long temp1 = 0; // Temperature of Battery A
+	long temp2 = 0; // Temperature of Battery B
+	int Pnum = 0;
+	int i = 13;
+
+	int index = 0;
+	uint8_t result;
+
+	/**************************** Monitor voltages, current, and Temperature of the batteries*********/
+	
+	
+	/*******************************************
+	 * Monitor voltage levels of batteries.    *
+	 * Use nodeVBattA and nodeVBattB to get    *
+	 * access to getVoltage() to get voltage   *
+	 * readings. Balance voltage levels of the *
+	 * batteries so that each battery has the  *
+	 * same voltage level. 
+	 ****************************************/
+	result = nodeVBattA.getVoltage(&BattVoltA);
+	if (result != ERROR_SUCCESS)
+	{
+		ERROR("Periodic", "Failed to get voltage of Battery A");
+		return result;
+	}
+	
+	result = nodeVBattB.getVoltage(&BattVoltB);
+	if (result != ERROR_SUCCESS)
+	{
+		ERROR("Periodic", "Failed to get voltage of Battery B");
+		return result;
+	}
+
+	if (BattVoltA > BattVoltB)
+	{
+		// if voltage of battery A is greater than the voltage of battery of B, redirect current to voltage B
+		// interate through nodesPVOut[] and set each one to go to battery B
+
+		while (nodesPVOut[index] < 8)
```

Comment:
Why not a contained for loop? BTW, this while loop is comparing a pointer to 8

Reply:
What do you mean by a contained loop? Like include this section in a single loop.